Cover new liquidity monitoring ACs + fix max OI not dropping (when expected) + new  🥒 step

witgaw opened the pull request:
Closes https://github.com/vegaprotocol/system-tests/issues/1095
Closes  #6387

witgaw committed:
refactor: add step to assure event(s) not emitted

witgaw committed:
chore: add AC codes

witgaw committed:
fix: max OI not dropping as expected

witgaw committed:
test: cover 0035-LIQM ACs

witgaw committed:
chore: run gofmt

witgaw reviewed (commented):
Ready for review.

witgaw committed:
chore: update CHANGELOG

EVODelavega commented on core/integration/steps/the_following_events_should_be_emitted.go:
```diff
@@ -32,6 +32,18 @@ func TheFollowingEventsShouldBeEmitted(broker *stubs.BrokerStub, table *godog.Ta
 	return nil
 }
 
+func TheFollowingEventsShouldNotBeEmitted(broker *stubs.BrokerStub, table *godog.Table) error {
```
This is a bit of an iffy step/check IMO. If we were to use this to check, say, that after a trade we don't enter auction, we'd have an error, as the events related to monitoring auction will still be returned by `broker.GetBatch`. I do think this could be a useful check to have, but perhaps we should qualify that we expect events of type X not to be emitted in the current/last block instead of in general

witgaw replied:
How is that different from the already existing `TheFollowingEventsShouldBeEmitted` step? It sort of requires manual work anyway (to clear events beforehand) as by default it contains all events from the beginning of the scenario.

witgaw replied:
We could at some point properly implement the `given`, `when`, `then` keywords (now they can just be used interchangeably) and then have all the events only cached for lookup during the course of a single `given`/`when` (+ optional `and`) clause, but that's clearly not something for this PR.

witgaw committed:
refactor: modify target stake test

Modify test:
1. Use block propagation rather than jumping to specific time points.
2. (as a result of 1.) shorten target stake window to avoid uncessary
block skipping
3. Extract common bits into background step

witgaw committed:
refactor: remove redundant variable

witgaw committed:
refactor: fix test

witgaw committed:
test: refactor to use blocks

+ cover a case of target stake dropping to 0

witgaw reviewed (commented):
Ready for another review

jeremyletang committed:
Merge branch 'develop' into liq-mon-acs
